product/views.py

```diff
@@ -1,3 +1,132 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+
+from product.models import MainCategory, SubCategory, Product
+
+class CategoryListView(View):
+    def get(self, request):
+        main_categories = MainCategory.objects.all()
+
+        result = [{
+            "id"   : main_category.id,
+            "name" : main_category.name,
+            "sub_categories"  : [{
+                "id"   : sub_category.id,
+                "name" : sub_category.name
+            } for sub_category in main_category.sub_categories.all()]
+        } for main_category in main_categories]
+
+        return JsonResponse({"main_categories" : result}, status = 200)
+
+class ProductListView(View): 
+    def get(self, request):
+        sub_category_id = int(request.GET.get('subcategoryid', None))
```

Comment:
snake case인 `subcategory_id`를 사용하던지 camel case인 `subcategoryId`를 사용하던지 둘 중에 하나로 수정해주세요.
```suggestion
        sub_category_id = int(request.GET.get('subcategoryId', None))
```

Reply:
URL에 들어갈 부분이락 대문자와 _를 쓰지 않게 되어 있는 것으로 알고 있는데
query parameter의 경우 상관없나요??